### PR TITLE
build system: port to bazel

### LINKED ISSUE #347: Upgrade to Bazel
We've migrated through multiple build systems, each far more usable than the last, but Glass is not cutting it, and we need something better -- and preferably something we don't have to develop ourselves.

So we're going with [Bazel](https://bazel.build/).

### PR DESCRIPTION
**Pull request by celskeggs**
This PR involves a port of the entire build system from Glass to Bazel. It is split up into individual commits that should walk you through the different parts to this. Intermediate commits should not be expected to be functional.

The only reason this patch is so big is because it includes a 61,276-line autogenerated kubernetes patch. It would be +5,914 / -7,287 otherwise, for a net removal of ~1400 lines.

Fixes #347.

---

Checklist:

 - [x] I have split up this change into one or more appropriately-delineated commits.
 - [x] The first line of each commit is of the form "[component]: do something"
 - [x] I have written a complete, multi-line commit message for each commit.
 - [x] I have formatted any Go code that I have changed with gofmt.
 - [x] I have signed each commit with my GPG key.
 - [x] My changes have passed CI.
 - [x] I have built my changes locally, and tested that the changes work as expected.
 - [x] I have deployed a cluster incorporating my changes, and checked that it deploys successfully.
 - [x] I have assigned this issue to an appropriate reviewer. (Choose @celskeggs if you are not otherwise certain.)
 - [x] I consider my PR complete and ready to be merged without my further input, assuming that it passes CI and code review.

### CODE REVIEW

**Comment by ichung** [comment]
:fire: :fire: :fire:

**Comment by krawthekrow** [comment]
Not critical, but it looks like zip_file.bzl (commit e0f3640 in the PR) provides an example for how you can define a rule that runs Python directly instead of bashing out, and would probably be a cleaner way to do template substitution.

**Comment by celskeggs** [comment]
Oh, yeah, it's absolutely possible to do the template substitution in starlark. I just haven't done the work to implement it like that yet.

File: platform/auth-monitor/auth-monitor.go [hunk 1]
```diff
@@ -11,16 +11,16 @@ import (
 	"crypto/tls"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sipb/homeworld/platform/keysystem/api"
+	"github.com/sipb/homeworld/platform/keysystem/api/reqtarget"
+	"github.com/sipb/homeworld/platform/keysystem/api/server"
+	"github.com/sipb/homeworld/platform/util/osutil"
```
Comment (ichung) [inline platform/auth-monitor/auth-monitor.go]: all go files: should organize commits into sections for clarity

Comment (celskeggs) [reply platform/auth-monitor/auth-monitor.go]: Will resolve this in a follow-up PR, for the sake of getting the core changes merged ASAP.

**Comment by celskeggs** [comment]
@krawthekrow can you re-review? I only added two commits.

**Comment by celskeggs** [comment]
Aww squid. It's broken because of the debian ISO issue. I'll fix this later.

**Pull request merged**